fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/QueryPlanInfo.java

```diff
@@ -141,4 +139,43 @@ public String toString() {
             return name;
         }
     }
+
+    /**
+     * A Builder for the {@link QueryPlanInfo} class. Since the plan info is immutable, this builder is used to populate
+     * the map with content. Once built, the {@link QueryPlanInfo} cannot be modified.
+     */
+    public static class Builder {
+        private final Map<QueryPlanInfoKey<?>, Object> infoMap;
+
+        private Builder() {
+            infoMap = new HashMap<>();
+        }
+
+        private Builder(QueryPlanInfo source) {
+            infoMap = new HashMap<>(source.info);
+        }
+
+        /**
+         * Set a value for the given key.
+         *
+         * @param key   the key to use
+         * @param value the value to associate with the key
+         * @param <T>   the type of the value to set (determined by the Key generic type)
+         * @return this
+         */
+        @Nonnull
+        public <T> Builder put(@Nonnull QueryPlanInfoKey<T> key, @Nonnull T value) {
+            infoMap.put(key, value);
+            return this;
+        }
+
+        @Nullable
+        public <T> T get(@Nonnull QueryPlanInfoKey<T> key) {
```

Comment:
Normally, builders don't have gets -- do you need this one?

Reply:
Yes, as it turns out, with the usage flow, we needed to log the contents of the PlanInfo before it is built, hence added a `get()`. It seemed to not interfere with immutability.